demisto_sdk/commands/common/docker/dockerhub_client.py

```diff
@@ -0,0 +1,382 @@
+        results_key: str = "results",
+    ):
+        """
+        Do a get request to dockerhub api
+
+        Args:
+            url_suffix: the URL suffix
+            next_page_url: the URL for the next page if pagination is required
+            headers: any custom headers
+            params: query parameters
+            results_key: the key to retrieve the results in case its a list
+        """
+        if url_suffix:
+            if not url_suffix.startswith("/"):
+                url_suffix = f"/{url_suffix}"
+            url = f"{self.docker_hub_api_url}{url_suffix}"
+        elif next_page_url:
+            url = next_page_url
+        else:
+            raise ValueError("either url_suffix/next_page_url must be provided")
+
+        response = self._session.get(
+            url,
+            headers=headers or {"Accept": "application/json"},
+            params=params or {"page_size": 1000} if not next_page_url else params,
+            verify=self.verify_ssl,
+            auth=(self.username, self.password)
+            if self.username and self.password
+            else None,
+        )
+        response.raise_for_status()
+        try:
+            raw_json_response = response.json()
+        except JSONDecodeError as e:
+            raise RuntimeError(
+                f"Failed to get response of {url=}, {response.text=}"
+            ) from e
+
+        amount_of_objects = raw_json_response.get("count")
+        if not amount_of_objects:
+            # received only a single record
+            return raw_json_response
+
+        logger.debug(f'Received {raw_json_response.get("count")} objects from {url=}')
+
+        results = raw_json_response.get(results_key) or []
+        if next_page_url := raw_json_response.get("next"):
+            results.extend(
+                self.do_docker_hub_get_request(
+                    next_page_url=next_page_url,
+                    headers=headers,
+                    params=params,
+                    results_key=results_key,
+                )
+            )
+
+        return results
+
+    @lru_cache
+    @retry(times=5, exceptions=(ConnectionError, Timeout))
+    def do_registry_get_request(
+        self,
+        url_suffix: str,
+        docker_image: str,
+        scope: DockerHubAuthScope = DockerHubAuthScope.PULL,
+        headers: Optional[Dict[str, Any]] = None,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """
+        Do a get request to a dockerhub registry
+
+        Args:
+            url_suffix: the URL suffix
+            docker_image: The docker-image name, e.g: demisto/pan-os-python
+            scope: what is the scope for the api-request.
+            headers: any custom headers
+            params: query parameters
+
+        Returns:
+
+        """
+        if not url_suffix.startswith("/"):
+            url_suffix = f"/{url_suffix}"
+
+        url = f"{self.registry_api_url}/{docker_image}{url_suffix}"
```

Comment:
Why not export all this code to a send request func? seems a bit repetitive

Reply:
I will make something that will do minimum code duplication